MekHQ/src/mekhq/campaign/mission/resupplyAndCaches/StarLeagueCache.java

```diff
@@ -0,0 +1,640 @@
+                default -> throw new IllegalStateException("Unexpected weight: " + weight);
+            };
+            case 4, 5 -> rollOutcome = switch (weight) {
+                case WEIGHT_LIGHT -> new int[]{WEIGHT_LIGHT, WEIGHT_LIGHT, WEIGHT_MEDIUM, WEIGHT_MEDIUM};
+                case WEIGHT_MEDIUM -> new int[]{WEIGHT_MEDIUM, WEIGHT_MEDIUM, WEIGHT_MEDIUM, WEIGHT_HEAVY};
+                case WEIGHT_HEAVY -> new int[]{WEIGHT_MEDIUM, WEIGHT_HEAVY, WEIGHT_HEAVY, WEIGHT_ASSAULT};
+                case WEIGHT_ASSAULT -> new int[]{WEIGHT_HEAVY, WEIGHT_ASSAULT, WEIGHT_ASSAULT, WEIGHT_ASSAULT};
+                default -> throw new IllegalStateException("Unexpected weight: " + weight);
+            };
+            case 6 -> rollOutcome = switch (weight) {
+                case WEIGHT_LIGHT -> new int[]{WEIGHT_LIGHT, WEIGHT_LIGHT, WEIGHT_MEDIUM, WEIGHT_HEAVY};
+                case WEIGHT_MEDIUM -> new int[]{WEIGHT_MEDIUM, WEIGHT_MEDIUM, WEIGHT_HEAVY, WEIGHT_HEAVY};
+                case WEIGHT_HEAVY -> new int[]{WEIGHT_HEAVY, WEIGHT_HEAVY, WEIGHT_HEAVY, WEIGHT_ASSAULT};
+                case WEIGHT_ASSAULT -> new int[]{WEIGHT_ASSAULT, WEIGHT_ASSAULT, WEIGHT_ASSAULT, WEIGHT_ASSAULT};
+                default -> throw new IllegalStateException("Unexpected weight: " + weight);
+            };
+            default -> throw new IllegalStateException("Unexpected value in getlanceWeights(): " + roll);
+        }
+
+        for (int outcome : rollOutcome) {
+            unitWeights.add(outcome);
+        }
+
+        return unitWeights;
+    }
+
+    public void createDudDialog(StratconTrackState track, StratconScenario scenario) {
+        StratconCoords stratconCoords = scenario.getCoords();
+
+        // Dialog dimensions and representative
+        final int DIALOG_WIDTH = 400;
+        final int DIALOG_HEIGHT = 200;
+
+        // Creates and sets up the dialog
+        JDialog dialog = new JDialog();
+        dialog.setTitle(resources.getString("dialog.title"));
+        dialog.setLayout(new BorderLayout());
+        dialog.setSize(UIUtil.scaleForGUI(DIALOG_WIDTH, DIALOG_HEIGHT));
+        dialog.setLocationRelativeTo(null);
+
+        // Defines the action when the dialog is being dismissed
+        ActionListener dialogDismissActionListener = e -> dialog.dispose();
+
+        // Associates the dismiss action to the dialog window close event
+        dialog.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent windowEvent) {
+                dialogDismissActionListener.actionPerformed(null);
+            }
+        });
+
+        // Prepares and adds the icon of the representative as a label
+        JLabel iconLabel = new JLabel();
+        iconLabel.setHorizontalAlignment(JLabel.CENTER);
+
+        ImageIcon speakerIcon = getSpeakerIcon(false);
+//        speakerIcon = scaleImageIconToWidth(speakerIcon, UIUtil.scaleForGUI(100));
```

Comment:
If you have the spoons, please delete the commented-out lines.

Reply:
So this class can be ignored in its entirety. It's initial work for a 50.03 feature. Resupplies and Caches got split up late in development, and it was easier to comment out the stuff that would cause problems than delete it and have to rewrite it later down the line